charmhub/errors.go

```diff
@@ -9,37 +9,20 @@ import (
 	"github.com/juju/juju/charmhub/transport"
 )
 
-// APIError extracts an API error from the given error or returns an error.
-func APIError(err error) (transport.APIError, error) {
-	if err == nil {
-		return transport.APIError{}, nil
-	}
-	if IsAPIError(err) {
-		return err.(transport.APIError), nil
-	}
-	return transport.APIError{}, errors.Annotatef(err, "not valid APIError")
-}
-
-// APIErrors extracts a slice of API errors from the given error or returns an
+// apiErrors extracts a slice of API errors from the given error or returns an
 // error.
-func APIErrors(err error) (transport.APIErrors, error) {
+func apiErrors(err error) (transport.APIErrors, error) {
 	if err == nil {
 		return nil, nil
 	}
-	if IsAPIErrors(err) {
+	if isAPIErrors(err) {
 		return err.(transport.APIErrors), nil
 	}
-	return nil, errors.Annotatef(err, "not valid APIErrors")
-}
-
-// IsAPIError checks to see if the error is a valid API error.
-func IsAPIError(err error) bool {
-	_, ok := errors.Cause(err).(transport.APIError)
-	return ok
+	return nil, errors.Annotatef(err, "not valid apiErrors")
 }
 
-// IsAPIErrors checks to see if the error is a valid series of API errors.
-func IsAPIErrors(err error) bool {
+// isAPIErrors checks to see if the error is a valid series of API errors.
+func isAPIErrors(err error) bool {
 	_, ok := errors.Cause(err).(transport.APIErrors)
 	return ok
 }
```

Comment:
We can use the recently added errors.Is() here right?

Reply:
Huh, I looked more closely at this and we don't need this code or `errors.Is` at all. The `handleBasicAPIErrors` function already takes a specific error type. So I've just removed/simplified this.